BasicGUI/Interpolator.h

```diff
@@ -0,0 +1,42 @@
+
+#ifndef INTERPOLATION_H
+#define INTERPOLATION_H
+
+#include <QColor>
+#include <QPixmap>
+#include <QPoint>
+#include <QVector>
+#include <cmath>
+
+/**
+ * @brief Instance of ColorColmponent Class represent single color component
+ * in RGB image format.
+ */
+//////////////////////////////////////////////////////////////////
+struct ColorComponent
+{
+    int iterator = 0;
+    int sum = 0;
+    int level = 0;
+};
+
+//////////////////////////////////////////////////////////////////
+class Interpolator : public QObject
+{
+    Q_OBJECT
+
+public:
+    Interpolator(QObject* parent = 0);
+
+    QColor interpolatorSetLedColor(QVector<QPointF> vector_points);
+    QVector<QPointF> interpolatorTransform(QPoint sr_obr, QRect rect,
```

Comment:
`Transform` could be a parameter, but we can leave for now.

Name of parameters are inconsistent between declaration and definition of method.

Reply:
fixed